pythia8.sh

```diff
@@ -4,14 +4,24 @@ source: https://github.com/alisw/pythia8
 requires:
   - LHAPDF
   - HepMC
+  - boost
 tag: alice/v8210
 ---
 #!/bin/sh
 cd $SOURCEDIR
+
+# jkl: boost headers are required for LHAPDF6
 ./configure --prefix=$INSTALLROOT \
             --enable-shared \
             --with-hepmc2=${HEPMC_ROOT} \
-            --with-lhapdf6=${LHAPDF_ROOT}
+            --with-lhapdf6=${LHAPDF_ROOT} \
+	    --with-boost=${BOOST_ROOT}
+
+if [[ $ARCHITECTURE =~ "slc5.*" ]]; then
+    ln -s LHAPDF5.h include/Pythia8Plugins/LHAPDF5.cc
+    ln -s LHAPDF6.h include/Pythia8Plugins/LHAPDF6.cc
+    sed -i -e 's#\$(CXX) -x c++ \$< -o \$@ -c -MD -w -I\$(LHAPDF\$\*_INCLUDE) \$(CXX_COMMON)#\$(CXX) -x c++ \$(<:.h=.cc) -o \$@ -c -MD -w -I\$(LHAPDF\$\*_INCLUDE) \$(CXX_COMMON)#' Makefile
+fi
```

Comment:
Leave it for now. We should try to see if upstream is willing to fix 
things in their code.

--
Ciao,
Giulio
On 27 Aug 2015, at 14:32, Jochen Klein wrote:

>> ./configure --prefix=$INSTALLROOT \
>>           --enable-shared \
>>           --with-hepmc2=${HEPMC_ROOT} \
>> -            --with-lhapdf6=${LHAPDF_ROOT}
>> +            --with-lhapdf6=${LHAPDF_ROOT} \
>> +	    --with-boost=${BOOST_ROOT}
>> +
>> +if [[ $ARCHITECTURE =~ "slc5.*" ]]; then
>> +    ln -s LHAPDF5.h include/Pythia8Plugins/LHAPDF5.cc
>> +    ln -s LHAPDF6.h include/Pythia8Plugins/LHAPDF6.cc
>> +    sed -i -e 's#\$(CXX) -x c++ \$< -o \$@ -c -MD -w 
>> -I\$(LHAPDF\$\*_INCLUDE) \$(CXX_COMMON)#\$(CXX) -x c++ \$(<:.h=.cc) 
>> -o \$@ -c -MD -w -I\$(LHAPDF\$\*_INCLUDE) \$(CXX_COMMON)#' Makefile
>> +fi
>
> This works around a compiler bug (feature?) that builds a precompiled 
> header instaed of an object file if called on a .h file despite being 
> told differently. I don't see why we should add workarounds on systems 
> for which they are not needed. But that's up to you.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/alisw/alidist/pull/64/files#r38089440